app/services/topic_bookmarkable.rb

```diff
@@ -12,7 +12,16 @@ def self.serializer
   end
 
   def self.preload_associations
-    [:topic_users, :tags, { posts: :user }]
+    [:tags, { posts: :user }]
+  end
+
+  def self.perform_custom_preload!(topic_bookmarks, guardian)
+    topics = topic_bookmarks.map(&:bookmarkable)
```

Comment:
I might be wrong here since I didn't run the code locally but I suspect this will cause an N+1 because `perform_custom_preload!` is being called before the `bookmarkable` association has been preloaded. Instead of `perform_custom_preload`, perhaps the hook should be after ``after_preload` instead which is called after the associations have been preloaded.

Reply:
Nice catch - pushed a fix to run custom preloading after the existing preload_associations